### PR TITLE
refactor: bashbot slack logic refactor

### PR DESCRIPTION
**Pull request by wisdommatt**
@mathew-fleisch I am currently refactoring the bashbot slack logic, I have changed function names and added comment/documentation to some of the codes. the new logic is in `internal/bashbot`. Am still in the process of refactoring the logic. Have been very busy lately, so just look at what i have done so far and drop your feedback/review.

### CODE REVIEW

**Comment by mathew-fleisch** [comment]
It is looking good so far! I like your changes. When you achieve parity, we can delete the old .go file. I've set up some automation in this branch that isn't working because you need to point at the new location (internal/slack/slack.go) here:
https://github.com/wisdommatt/bashbot/blob/cobra-cli/Makefile#L6

Once you do, it will build and test the changes in a KinD cluster within github actions. I hope you can see the console-log/output from the automation by clicking the 'Details' links next to the failed tests `PR Tests / Lint and Unit Tests (pull_request)` and `PR Tests / KinD Integration Tests (pull_request)`

**Comment by wisdommatt** [comment]
> It is looking good so far! I like your changes. When you achieve parity, we can delete the old .go file. I've set up some automation in this branch that isn't working because you need to point at the new location (internal/slack/slack.go) here: https://github.com/wisdommatt/bashbot/blob/cobra-cli/Makefile#L6
> 
> Once you do, it will build and test the changes in a KinD cluster within github actions. I hope you can see the console-log/output from the automation by clicking the 'Details' links next to the failed tests `PR Tests / Lint and Unit Tests (pull_request)` and `PR Tests / KinD Integration Tests (pull_request)`
> 
> Maybe we can split out the struct definitions to a separate file?

Splitting the struct definitions to a separate file will be a good idea

File: LICENSE [hunk 1]
```diff
@@ -1,21 +1,202 @@
...
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
+
+                                 Apache License
```
Comment (mathew-fleisch) [inline LICENSE:2]: Why did you change the license?

**Comment by mathew-fleisch** [comment]
Hey @wisdommatt can you rebase your branch? I'd like to finally get this merged, but it is a bit out of date now.

File: .tool-versions [hunk 2]
```diff
@@ -1,4 +1,15 @@
-helm 3.7.2
-kubectl 1.23.2
-kustomize 4.4.1
+awscli 2.4.27
```
Comment (mathew-fleisch) [inline .tool-versions:1]: please use the .tool-versions from the v2 branch
